### PR TITLE
ext: mbedtls: Extend mbedtls generic config

### PR DESCRIPTION
**Pull request by rlubos**
A follow up of discussion in #12063.

This PR extends the generic config file for mbedTLS with additional configuration options, including the following:
* More algorithms to enable,
* Add switch to enable all algorithms in specific group,
* Add RNG section,
* CTR_DRBG is no longer always enabled when AES is enabled. Now there is a config to enable/disable it with dependency to AES.

Additionally, second commits unifies prefixes for mbedTLS Kconfig options. Now all options (including those in generic config file) have `MBEDTLS_` prefix.

### CODE REVIEW

File: ext/lib/crypto/mbedtls/Kconfig.tls-generic [hunk 1]
```diff
@@ -135,16 +181,45 @@ config TLS_CIPHER_DES_ENABLED
 	bool "Enable the DES block cipher"
 	default y
 
+config TLS_CIPHER_ARC4_ENABLED
+	bool "Enable the ARCFOUR stream cipher"
```
Comment (pfalcon) [inline ext/lib/crypto/mbedtls/Kconfig.tls-generic]: Well, it's still "ARC4", I'd say. Actually, it's RC4 in normal speak even, where A in "ARC4" is for "alleged".

File: ext/lib/crypto/mbedtls/Kconfig.tls-generic [hunk 2]
```diff
@@ -135,16 +181,45 @@ config TLS_CIPHER_DES_ENABLED
...
+
+config TLS_CIPHER_BLOWFISH_ENABLED
+	bool "Enable the Blowfish block cipher"
+
 config TLS_CIPHER_CCM_ENABLED
 	bool "Enable the Counter with CBC-MAC (CCM) mode for 128-bit block cipher"
 	depends on TLS_CIPHER_AES_ENABLED || TLS_CIPHER_CAMELLIA_ENABLED
 
+config TLS_CIPHER_MODE_XTS_ENABLED
+	bool "Enable Xor-encrypt-xor with ciphertext stealing mode (XTS) for AES"
```
Comment (pfalcon) [inline ext/lib/crypto/mbedtls/Kconfig.tls-generic]: And here I'd say it goes just too detailed - for layman, both "XTS" and "Xor-encrypt-xor with ciphertext stealing" mean about the same - weird crypto mumbo-jumbo ;-). No need to change now that the work is done, the effort to educate people on spot is always appreciated of course.

File: ext/lib/crypto/mbedtls/Kconfig.tls-generic [hunk 3]
```diff
@@ -160,8 +235,30 @@ config TLS_MAC_SHA256_ENABLED
...
+	bool "Enable the CTR_DRBG AES-256-based random generator"
+	depends on TLS_CIPHER_AES_ENABLED
+	default y
+
+config TLS_HMAC_DRBG_ENABLED
+	bool "Enable the HMAC_DRBG random generator"
+
+comment "Other configurations"
+
+config TLS_CHACHAPOLY_AEAD_ENABLED
```
Comment (pfalcon) [inline ext/lib/crypto/mbedtls/Kconfig.tls-generic]: Well, and this is kinda misplaced under "Other configurations", it should go together with similar AEAD ciphers, e.g. TLS_CIPHER_CCM_ENABLED.

**Comment by pfalcon** [comment]
Great response to challenges of https://github.com/zephyrproject-rtos/zephyr/pull/12063, I'd suggest some tweaks to Kconfig option ordering/wording.

File: ext/lib/crypto/mbedtls/Kconfig.tls-generic (hunk 2 above)
Comment (rlubos) [reply ext/lib/crypto/mbedtls/Kconfig.tls-generic]: Just to clarify, this "mumbo-jumbo" is actually original description from mbedTLS config file template.

File: ext/lib/crypto/mbedtls/Kconfig.tls-generic (hunk 3 above)
Comment (rlubos) [reply ext/lib/crypto/mbedtls/Kconfig.tls-generic]: Agreed, will move

**Comment by rlubos** [comment]
Applied fixes according to the review.

**Review by pfalcon — APPROVED** [review]
I'd say commits should have been applied in the reverse order - it's a bit strange that you introduce e.g. TLS_CIPHER_CHACHA20_ENABLED just to change it to MBEDTLS_CIPHER_CHACHA20_ENABLED in the next commit.

But elaborating this stuff is definitely better than not, so let's be grateful it's done at all ;-). Thanks!

**Comment by rveerama1** [comment]
LGTM, thanks.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**